File: my-folder/0078-subsets/solution.cpp
class Solution {
public:
    void findSubset(vector<int>& nums,vector<vector<int>>& ans,vector<int> temp, int ind){
        ans.push_back(temp);
        for(int i=ind;i<nums.size();i++){
            temp.push_back(nums[i]);
            findSubset(nums,ans,temp,i+1);
            temp.pop_back();
        }

    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans ;
        findSubset(nums,ans,{},0);
        return ans;

        
    }
};

File: my-folder/0078-subsets/solution.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> ans = new ArrayList<>();
        int subsetLength = 1<<nums.length;
        for(int num=0;num<subsetLength;num++){
            List<Integer> temp = new ArrayList<>();
            for(int i = 0; i<nums.length;i++){
                if((num&(1<<i))!=0) temp.add(nums[i]);
            }
            ans.add(temp);
        }
        return ans;
    }
}
